### PR TITLE
fix(toast): use flex

### PR DESCRIPTION
**Pull request by gergelyke**
closes https://github.com/uber-web/baseui/issues/1065

### CODE REVIEW

**Comment by chasestarr** [comment]
we've generally avoided grid so far, unless opted-in by developers since it doesn't have full support from older browsers. could this be refactored to use flex?

**Comment by gergelyke** [comment]
sure, I'll do that

**Comment by gergelyke** [comment]
@chasestarr updated, what do you think?

File: src/toast/toast.js [hunk 1]
```diff
@@ -185,9 +186,13 @@ class Toast extends React.Component<ToastPropsT, ToastPrivateStateT> {
                 overrides={closeIconOverrides}
               />
             ) : null}
-            {typeof children === 'function'
-              ? children({dismiss: this.dismiss})
-              : children}
+            {typeof children === 'function' ? (
+              children({dismiss: this.dismiss})
+            ) : (
+              <Block display="flex" flex="1">
```
Comment (nadiia) [inline src/toast/toast.js]: Could we avoid using this additional wrapper? It's not in overrides so created an element that can't be customized.

Comment (chasestarr) [reply src/toast/toast.js]: yeah, probably just add 'flex' to `Body` with `justifyItems: 'space-between'` not totally sure what's going on with the markup ordering in this component. I imagine re-order close icon and children would avoid the `order` css property

**Comment by chasestarr** [comment]
check out this sandbox with a way to position without adding a new element https://codesandbox.io/s/382vlqyyqq you'll need to change the ordering of the close icon and children in the component file so that it renders in the correct order without css

**Comment by nadiia** [comment]
> check out this sandbox with a way to position without adding a new element https://codesandbox.io/s/382vlqyyqq you'll need to change the ordering of the close icon and children in the component file so that it renders in the correct order without css

Right, rendering it after the content makes sense. Let's move it @gergelyke. The wrapper element may still be useful if there are more than one children.

**Comment by gergelyke** [comment]
@chasestarr @nadiia updated, what do you think?

File: src/toast/toast.js [hunk 2]
```diff
@@ -169,6 +176,13 @@ class Toast extends React.Component<ToastPropsT, ToastPrivateStateT> {
             onMouseEnter={this.onMouseEnter}
             onMouseLeave={this.onMouseLeave}
           >
+            {typeof children === 'function' ? (
+              children({dismiss: this.dismiss})
+            ) : (
+              <ToastContent {...sharedProps} {...toastContentProps}>
```
Comment (nadiia) [inline src/toast/toast.js]: Does it make sense to wrap the children in both cases? `function` | React.Children

**Comment by chasestarr** [comment]
my issue is with the new `ToastContent` override. can't we get the proper outcome without it?

**Pull request merged**